pkg/kubelet/kuberuntime/logs/logs.go

```diff
@@ -358,7 +359,11 @@ func ReadLogs(ctx context.Context, path, containerID string, opts *LogOptions, r
 					return err
 				}
 				if recreated {
```

Comment:
Isn't `rectreated` is set to `true` means that the file exists? If it's set to `true` if file isn't exist then probably `waitLogs` should be fixed.

Reply:
If `recreated` is `true` means the log file has beed changed(including being created, renamed and removed), and if log rotation is in progress, CRI will recreate it. 

`recreated` may not be entirely precise here.

Reply:
waitLogs descriptions states that `recreated` is set to `true` if file is re-created:
```
waitLogs wait for the next log write. It returns two booleans and an error. The first boolean indicates whether a new log is found; the second boolean if the log file was recreated; the error is error happens during waiting new logs.
```
If it's not the case then it probably should be fixed.

My point is that instead of working around the case when file doesn't exist it's better to fix `waitLogs` to return correct info.

BTW, `waitLog` also returns `found`, which means that the new log is found. You can check if `found` is set to `true` instead of trying to open the file 8 times.

Reply:
You are correct, we should fix waitLogs instead of retryOpen.

Reply:
> @icefed Please, reimplement according to what we've discussed [here](https://github.com/kubernetes/kubernetes/pull/118500#pullrequestreview-1467066749)

@bart0sh `recreated` now returns the correct value. Is there anything else that needs to be modified? Do you mean removing `retrying`?

Reply:
If we returning `false, false, nil` on fsnotify.Rename, fsnotify.Create event will not be received.

Reply:
More modifications need to be made in ReadLogs to accomplish this.

Reply:
> If we returning false, false, nil on fsnotify.Rename, fsnotify.Create event will not be received.

That means that fsnotify watcher doesn't watch logs directory, I guess. It probably should. Or it should be reconfigured to watch new file name after renaming.

Reply:
Watching logs directory is not ideal as log rotation also involves compression operations.  Reconfigures to new log file is not enough to fix the bug, more modifications need to be made in ReadLogs.

If both `found` and `recreated` are true, `ReadLogs` will return and cannot proceed to the next `waitLogs`.

Reply:
I've made some additional modifications and added unit tests.